pkg/types/types.go

```diff
@@ -197,29 +197,59 @@ type Metric struct {
 }
 
 // MergeMetrics merges metrics by name.
-func MergeMetrics(metrics [][]Metric) []Metric {
+// It returns merged metrics, number of rendered data points for the returned metrics,
+// and number of inconsistent data points seen (if consistencyCheck is true).
+func MergeMetrics(metrics [][]Metric, consistencyCheck bool) ([]Metric, int, int) {
 	if len(metrics) == 0 {
-		return nil
+		return nil, 0, 0
 	}
 
 	if len(metrics) == 1 {
-		return metrics[0]
+		pointCount := 0
+		ms := metrics[0]
+		for _, m := range ms {
+			pointCount += len(m.Values)
+		}
+		return ms, pointCount, 0
 	}
 
-	names := make(map[string][]Metric)
+	metricByNames := make(map[string][]Metric)
 
 	for _, ms := range metrics {
 		for _, m := range ms {
-			names[m.Name] = append(names[m.Name], m)
+			metricByNames[m.Name] = append(metricByNames[m.Name], m)
 		}
 	}
 
 	merged := make([]Metric, 0)
-	for _, ms := range names {
-		merged = append(merged, mergeMetrics(ms))
+	pointCount := 0
+	inconsistencyCount := 0
+	inconsMetricReportLimit := 10
```

Comment:
nit: imho this var is not really necessary, as the meaning of hardcoded value would be quite clear as well

Reply:
It is necessary to limit the volume of the logs. But I agree that hard-coding is not great here. Please move to config.

Reply:
I think moving it to config is an overkill as we probably will not configure it at any points in the future.
I will go for a constant in the same file. Any thoughts?

Reply:
I agree that it doesn't seem like something we would need to configure. Also we could it in the future, if it will be needed. IMHO we don't even need the constant, but could just inline the value, as the meaning would be pretty clear from the code.

@bom-d-van wdyt?

Reply:
I'm ok with a named constant. actually it's even better to declare it as `const` if it's truly a constant.

```
const inconsMetricReportLimit = 10
```